### DISCUSSION TITLE
Testing with concurrent rate limiters

### DISCUSSION THREAD

**Discussion by wheresrobert** [sidekiq/sidekiq#5348] [Q&A]
Hello,

What is the recommended way to test code that uses a concurrent rate limiter?

Since `Sidekiq::Testing.inline!` runs the jobs immediately, the rate limit is hit immediately, throwing an exception, which then fails the test.

How do I work around this?

**Comment by mperham** [comment]
I have no suggestions beyond what is in the wiki:

https://github.com/mperham/sidekiq/wiki/Ent-Rate-Limiting#testing

**Reply by wheresrobert** [reply]
Thank you. 

I've stubbed out the concurrent limiter in the test suite. I'll trust that Sidekiq is working as is described. The exceptions suggest that it is.
